pyriemann/spatialfilters.py

```diff
@@ -467,3 +469,266 @@ def fit(self, X, y):
+
+        # whitening, Eq.(8) in [2]
+        whit_filters = pca_filters @ numpy.diag(1. / numpy.sqrt(pca_vals))
+        whit_inv_filters = pca_filters @ numpy.diag(numpy.sqrt(pca_vals))
+
+        # apply dimension reduction and whitening on cospectra
+        cosp_rw = whit_filters.T @ self._cosp_channels @ whit_filters
+
+        # approximate joint diagonalization, currently by Pham's algorithm [3]
+        diag_filters, self._cosp_sources = ajd_pham(
+            cosp_rw,
+            n_iter_max=100,
+            sample_weight=self._weights)
+
+        # computation of forward and backward filters, Eq.(9) and (10) in [2]
+        self.forward_filters_ = diag_filters @ whit_filters.T
+        self.backward_filters_ = whit_inv_filters @ inv(diag_filters)
+        return self
+
+    def transform(self, X):
+        """Transform channel space to source space, applying forward spatial
+        filters.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_trials, n_channels, n_samples)
+            ndarray of trials in channel space.
+
+        Returns
+        -------
+        source : ndarray, shape (n_trials, n_sources, n_samples)
+            ndarray of trials in source space.
+        """
+        if X.ndim != 3:
+            raise ValueError('X must have 3 dimensions')
+        if X.shape[1] != self.n_channels_:
+            raise ValueError('X has not the good number of channels')
+
+        source = self.forward_filters_ @ X
+        return source
+
+    def inverse_transform(self, X, supp=None):
+        """Transform source space to channel space, applying backward spatial
+        filters, with the possibility to suppress some sources, like in BSS
+        filtering/denoising.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_trials, n_sources, n_samples)
+            ndarray of trials in source space.
+        supp : list of int | None, (default None)
+            List of indices of sources to suppress.
+
+        Returns
+        -------
+        signal : ndarray, shape (n_trials, n_channels, n_samples)
+            ndarray of trials in channel space.
+        """
+        if X.ndim != 3:
+            raise ValueError('X must have 3 dimensions')
+        if X.shape[1] != self.n_sources_:
+            raise ValueError('X has not the good number of sources')
+
+        denois = numpy.eye(self.n_sources_)
+        if supp is None:
+            pass
+        elif isinstance(supp, list):
+            for s in supp:
+                denois[s, s] = 0
+        else:
+            raise ValueError('Parameter supp must be a list of int, or None')
+
+        signal = self.backward_filters_ @ denois @ X
+        return signal
+
+    def get_src_expl_var(self, X):
+        """Estimate explained variances of sources, Appendix D in [1].
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_trials, n_channels, n_samples)
+            ndarray of trials in channel space.
```

Comment:
```suggestion
            Trials in channel space.
```
basically don't repeat the type when it's written above